a1/src/main/java/ro/utcn/spet/a1/repository/memory/InMemoryRepositoryFactory.java

```diff
@@ -0,0 +1,31 @@
+package ro.utcn.spet.a1.repository.memory;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+import ro.utcn.spet.a1.repository.api.QuestionRepository;
+import ro.utcn.spet.a1.repository.api.RepositoryFactory;
+import ro.utcn.spet.a1.repository.api.TagRepository;
+import ro.utcn.spet.a1.repository.api.UserRepository;
+
+@Component
+@ConditionalOnProperty(name = "a1.repository-type", havingValue = "MEMORY")
+public class InMemoryRepositoryFactory implements RepositoryFactory {
+    private final InMemoryQuestionRepository repositoryQ = new InMemoryQuestionRepository();
```

Comment:
Not the best naming convention in the world. Consider renaming to `questionRepository`, `userRepository`, etc.